app/code/core/Mage/Checkout/Model/Type/Onepage.php

```diff
@@ -730,7 +730,7 @@ protected function _prepareNewCustomerQuote()
 
         Mage::helper('core')->copyFieldset('checkout_onepage_quote', 'to_customer', $quote, $customer);
         $customer->setPassword($customer->decryptPassword($quote->getPasswordHash()));
-        $passwordCreatedTime = $this->_checkoutSession->getData('_session_validator_data')['session_expire_timestamp']
+        $passwordCreatedTime = $_SESSION[Mage_Core_Model_Session_Abstract_Varien::VALIDATOR_KEY]['session_expire_timestamp']
```

Comment:
Maybe it would be possible to add a `getValidatorData()` method to the `Mage_Core_Model_Session_Abstract_Varien` class? That would be inherited by all sessions, so the above line could become something like:
`$this->_checkoutSession->getValidatorData()`
Internally such a method would still access the $_SESSION superglobal, but that seems acceptable to me since it would be in a class whose purpose is to deal with the session.

Reply:
sounds good for me

Reply:
Ok. I'll update this PR ASAP.

Reply:
@Flyingmana I've updated the PR. Unforntunately the "getValidatorData" name was already used, so I had to call the new method "getValidatorResult". I hope it's fine.

Reply:
Why is this implemented differently than i described in the bug ticket?
All other occurrences of setPasswordCreatedAt() use time() and nothing session-related.

Reply:
@imaphpdev [It was made like that in 1.9.4.4 by Magento](https://github.com/OpenMage/magento-mirror/blob/magento-1.9/app/code/core/Mage/Checkout/Model/Type/Onepage.php#L732). My guess is that they had a security bug to fix in the checkout and they only changed the code where the bug occurred, and didn't bother to update all other places in the same way.
Just my guess, though.

Reply:
From what I see, the result of `getValidatorData()` is assigned to `$_SESSION[self::VALIDATOR_KEY]`:
https://github.com/OpenMage/magento-lts/blob/447e27ed999ac262d64528038195114eec4f64a4/app/code/core/Mage/Core/Model/Session/Abstract/Varien.php#L461-L462

So since the newly added method will be retrieving the result of `getValidatorData()` from the session, it makes sense to me to add the "Session" prefix to the name.

Reply:
Done